utils/funcexp/func_reverse.cpp

```diff
@@ -73,13 +74,15 @@ std::string Func_reverse::getStrVal(rowgroup::Row& row,
     // string assignment and implement a ref-count. Reversing in the
     // string buffer has the affect of reversing all strings from
     // which this one derived.
-    int len = str.length();
-    char* pbuf = new char[len + 1];
-    strncpy(pbuf, str.c_str(), len);
-    pbuf[len] = 0;
-    char* end = pbuf + len - 1;
+    wstring wstr = utf8::utf8_to_wstring(str);
+    int wlen = wstr.length();
+    wchar_t* pbuf = new wchar_t[wlen + 1];
+    wcsncpy(pbuf, wstr.c_str(), wlen);
```

Comment:
I think we should avoid this copy operation sending down 4 pointers into reverse(). begin and end pointers for the source string and two similar for the destination string.